# Description: Compile in our own PowerSaveBlocker implementation that
#  can delegate to Oxide
# Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/content/content_browser.gypi b/content/content_browser.gypi
--- a/content/content_browser.gypi
+++ b/content/content_browser.gypi
@@ -56,16 +56,17 @@
     '../third_party/WebKit/public/blink_headers.gyp:blink_headers',
     # The public render_widget_host.h needs to re-export skia defines.
     '../skia/skia.gyp:skia',
     'content_common_mojo_bindings.gyp:content_common_mojo_bindings',
   ],
   'include_dirs': [
     '..',
     '<(INTERMEDIATE_DIR)',
+    '<(DEPTH)/../../..',
   ],
   'variables': {
     'public_browser_sources': [
       'public/browser/access_token_store.h',
       'public/browser/android/browser_media_player_manager_register.cc',
       'public/browser/android/browser_media_player_manager_register.h',
       'public/browser/android/compositor.h',
       'public/browser/android/compositor_client.h',
@@ -1118,17 +1119,17 @@
       'browser/power_monitor_message_broadcaster.cc',
       'browser/power_monitor_message_broadcaster.h',
       'browser/power_save_blocker_android.cc',
       'browser/power_save_blocker_android.h',
       'browser/power_save_blocker_chromeos.cc',
       'browser/power_save_blocker_impl.cc',
       'browser/power_save_blocker_impl.h',
       'browser/power_save_blocker_mac.cc',
-      'browser/power_save_blocker_ozone.cc',
+      #'browser/power_save_blocker_ozone.cc',
       'browser/power_save_blocker_win.cc',
       'browser/power_save_blocker_x11.cc',
       'browser/power_usage_monitor_impl.cc',
       'browser/power_usage_monitor_impl.h',
       'browser/presentation/presentation_service_impl.cc',
       'browser/presentation/presentation_service_impl.h',
       'browser/presentation/presentation_type_converters.cc',
       'browser/presentation/presentation_type_converters.h',
@@ -1595,16 +1596,18 @@
       'browser/webui/web_ui_impl.cc',
       'browser/webui/web_ui_impl.h',
       'browser/webui/web_ui_message_handler.cc',
       'browser/zygote_host/zygote_host_impl_linux.cc',
       'browser/zygote_host/zygote_host_impl_linux.h',
       'zygote/zygote_linux.cc',
       'zygote/zygote_linux.h',
       'zygote/zygote_main_linux.cc',
+      '<(DEPTH)/../../../shared/port/content/browser/power_save_blocker_oxide.cc',
+      '<(DEPTH)/../../../shared/port/content/browser/power_save_blocker_oxide.h',
     ],
     'android_browser_sources': [
       'browser/android/java/gin_java_bound_object.cc',
       'browser/android/java/gin_java_bound_object.h',
       'browser/android/java/gin_java_bound_object_delegate.cc',
       'browser/android/java/gin_java_bound_object_delegate.h',
       'browser/android/java/gin_java_bridge_dispatcher_host.cc',
       'browser/android/java/gin_java_bridge_dispatcher_host.h',
